pandas/indexes/base.py

```diff
@@ -162,20 +165,25 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None,
         # index-like
         elif isinstance(data, (np.ndarray, Index, ABCSeries)):
 
-            if (issubclass(data.dtype.type, np.datetime64) or
-                    is_datetimetz(data)):
+            if is_datetime64_any_dtype(data):
                 from pandas.tseries.index import DatetimeIndex
                 result = DatetimeIndex(data, copy=copy, name=name, **kwargs)
```

Comment:
maybe make a function to do this:

```
if is_datetime64_any_dtype(data):
     from pandas.tseries.index import DatetimeIndex
     return DatetimeIndex._from_index(data, copy=copy, name=name, **kwargs)
....
# timedelta
# period
```

IOW make the coercion happen inside the ``_from_index`` class constructor